scripts/jenkins/Jenkinsfile

```diff
@@ -278,67 +278,100 @@ def buildImageStage(image) {
  */
 def generateMainBuildStages() {
     def tasks = [:]
+
+    // Add a task that should build the whole project to catch all test errors
+    // in a standard environment
     tasks << buildAndTest(
         "debian-stable-full",
         DOCKER_IMAGES.stretch,
         CMAKE_FLAGS_BUILD_ALL +
             CMAKE_FLAGS_COVERAGE,
         [TEST.ALL, TEST.MEM, TEST.INSTALL]
     )
+
+    // Check the ABI and API compatibility of Elektra
     tasks << buildIcheck()
+
+    // Check if release notes have been updated
     tasks << buildCheckReleaseNotes()
+
     return tasks
 }
 
 /* Generate Test stages for full test coverage
  */
 def generateFullBuildStages() {
     def tasks = [:]
+
+    // Run the open tasks plugin and sloccount
     tasks << buildTodo()
+
+    // Build doc and upload
     tasks << buildDoc()
+
+    // Build Elektra with ASAN enabled
+    // Detects memory leaks
     tasks << buildAndTestAsan(
         "debian-stable-asan",
         DOCKER_IMAGES.stretch,
         CMAKE_FLAGS_BUILD_ALL
     )
+
+    // Build Elektra with clang and ASAN
+    // Detects memory leaks
```

Comment:
with ASAN. In which jobs do we run valgrind?

Reply:
In a lot of them, each one that has the TEST.MEM flag set. Should I mention it explicitly or is it visible enough?